nodebuilder/share/config.go

```diff
@@ -36,6 +38,7 @@ type Config struct {
 func DefaultConfig() Config {
 	return Config{
 		PeersLimit:        5,
+		DialTimeout:       time.Second * 60,
```

Comment:
Is there any possible problems of dialing takes longer than discovery interval? Like triggering extra discovery for example 

Do we have any idea behind timeout having this value? If so would be nice to add comment section explaining why this value is default.

Reply:
> Is there any possible problems of dialing takes longer than discovery interval? Like triggering extra discovery for example

No, as long as we keep waitgroup, we won't start a new discovery until we stop the previous one

This value is exactly the same as it is in libp2p for the actual low-level dial. Have no reasoning for why it is 1min, could be two as well. It should be just big enough for the Connect so that It can perform internal address discoveries.

Reply:
> Does the waitgroup sync means that if one peer fails to connect, discovery will need to wait until all connection attempts finish before looking for replacement peer?

No, because FindPeer returned channel still gives us new values.